File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/ApolloCall.java
package com.apollographql.apollo3.runtime.java;

import com.apollographql.apollo3.api.Operation;
import com.apollographql.apollo3.runtime.java.interceptor.ApolloDisposable;
import org.jetbrains.annotations.Nullable;

public interface ApolloCall<D extends Operation.Data> {
  /**
   * Schedules the request to be executed at some point in the future.
   *
   * @param callback Callback which will handle the response or a failure exception.
   */
  ApolloDisposable enqueue(@Nullable ApolloCallback<D> callback);
}

File: apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo3/ApolloCall.kt
package com.apollographql.apollo3

import com.apollographql.apollo3.api.Operation
import com.apollographql.apollo3.api.ApolloResponse
import kotlinx.coroutines.flow.Flow

interface ApolloCall<D: Operation.Data> {
    fun execute(): Flow<ApolloResponse<D>>
}

interface ApolloQueryCall<D: Operation.Data> : ApolloCall<D>

interface ApolloMutationCall<D: Operation.Data> : ApolloCall<D>

interface ApolloSubscriptionCall<D: Operation.Data> : ApolloCall<D>
